tests/loan-token/tokenFunctionality.test.js

```diff
@@ -113,6 +128,146 @@ contract("LoanTokenFunctionality", (accounts) => {
+                staking.address,
+                tokenOwner,
+                cliff,
+                cliff,
+                feeSharingCollector.address
+            );
+
+            vestingInstance = await VestingLogic.at(vestingInstance.address);
+            // important, so it's recognized as vesting contract
+            await staking.addContractCodeHash(vestingInstance.address);
+
+            const previousTokenOwnerBalance = await loanToken.balanceOf(tokenOwner);
+            const previousVestingContractBalance = await loanToken.balanceOf(
+                vestingInstance.address
+            );
+
+            // transfer the tokens to the vesting contract address
+            await loanToken.transfer(vestingInstance.address, amount_sent);
+
+            const latestTokenOwnerBalance = await loanToken.balanceOf(tokenOwner);
+            const latestVestingContractBalance = await loanToken.balanceOf(
+                vestingInstance.address
+            );
+
+            /** Token owner of the contract should receive the amount */
+            expect(previousTokenOwnerBalance.add(amount_sent).toString()).to.equal(
+                latestTokenOwnerBalance.toString()
+            );
+
+            /** Vesting contract should not receive the amount */
+            expect(
+                previousVestingContractBalance.add(latestVestingContractBalance).toString()
+            ).to.equal("0");
+        });
+
+        it("should transfer to the passed recipient if passed recipient is EOA / Contract that has no tokenOwner() function", async () => {
+            // Creating the Staking Instance (Staking Modules Interface).
+            const stakingProxy = await StakingProxy.new(SOVToken.address);
+            const modulesObject = await getStakingModulesObject();
+            const staking = await deployAndGetIStaking(stakingProxy.address, modulesObject);
+
+            await loanToken.setStakingContractAddress(staking.address);
+            expect(await loanToken.getStakingContractAddress()).to.equal(staking.address);
+
+            // we register the contract that has no tokenOwner function
+            const dummyContractAddress = stakingProxy.address;
+            await staking.addContractCodeHash(dummyContractAddress);
+
+            const previousDummyContractBalance = await loanToken.balanceOf(dummyContractAddress);
+
+            // transfer the tokens to the contract that has no tokenOwner function
+            await loanToken.transfer(dummyContractAddress, amount_sent);
+            const latestDummyContractBalance = await loanToken.balanceOf(dummyContractAddress);
+
+            /** The dummy contract should receive the token */
+            expect(previousDummyContractBalance.add(amount_sent).toString()).to.equal(
+                latestDummyContractBalance.toString()
+            );
+        });
+
+        it("should transfer to the passed recipient if the passed recipient is not vesting contract", async () => {
```

Comment:
isn't it already tested in the previous test: EOA and a random contract, not registered as a VestingContract?

Reply:
The previous test is more about the EOA and random contract but registered as VestingContrac (AC No.2)
But for this test, it's more to test the EOA that not registered as VestingContract (AC No.3)